sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -76,14 +93,42 @@ namespace Azure { namespace Core {
         std::string const& timeString,
         DateFormat format = DateFormat::Rfc1123);
 
+  private:
     /**
      * @brief Get a string representation of the @DateTime.
      *
      * @param format The representation format to use.
+     * @param fractionFormat The format for the fraction part of the Datetime. Only supported by
+     * ISO8601
      *
      * @throw DateTimeException If year exceeds 9999, or if \p format is not recognized.
      */
-    std::string ToString(DateFormat format = DateFormat::Rfc1123) const;
+    std::string ToString(DateFormat format, TimeFractionFormat fractionFormat) const;
+
+  public:
+    /**
+     * @brief Get a string representation of the @DateTime.
+     *
+     * @param format The representation format to use.
+     *
+     * @throw DateTimeException If year exceeds 9999, or if \p format is not recognized.
+     */
+    std::string ToString(DateFormat format = DateFormat::Rfc1123) const
+    {
+      return ToString(format, TimeFractionFormat::DropTrailingZeros);
+    };
+
+    /**
+     * @brief Get a string representation of the @DateTime formated with ISO8601.
+     *
+     * @param fractionFormat The format that is applied to the fraction part from the ISO8601 date.
+     *
+     * @throw DateTimeException If year exceeds 9999, or if \p fractionFormat is not recognized.
+     */
+    std::string ToIso8601String(TimeFractionFormat fractionFormat) const
```

Comment:
Having a `DateFormat` specific API to convert to string defeats the purpose of that enum.

We should have a single `ToString()` method that takes a `format` that encapsulates any future customizations.

When we add a new `DateFormat`, we don't want to add a new `ToFooString()` method as well.

Reply:
> It is very unlikely to keep adding more formats for the date.

If you look at .NET's DateTime, I would be surprised if there aren't further requests for **a lot more** formats in the future.

Reply:
Remember we are not a language framework.
The reason we would add a new format is only if one Azure service requires it for any rest api

Reply:
I understand. Other Azure services in the future will likely need more formats such as trailing/leading whitespaces or if timezone info is present or not). For example, look at these:
https://github.com/Azure/azure-sdk-for-net/blob/dc23a39caf37d07f02512540d20c29866ff83759/sdk/insights/Microsoft.Azure.Management.Insights/tests/Helpers/JsonExtensions.cs#L28
https://github.com/Azure/azure-sdk-for-net/blob/e95f06fbaf015fb7ca41633b5032c37f5e73cfee/sdk/search/Azure.Search.Documents/src/SearchDocument/SearchDocument.Core.cs#L130
https://github.com/Azure/azure-sdk-for-net/blob/2f73639327af22ca75399e0f632215988d317db2/sdk/storage/Azure.Storage.Blobs.Batch/src/Shared/HttpRuleParser.cs#L250